### DISCUSSION TITLE
Installation of libraries through sudo and usage of GPUs

### DISCUSSION THREAD

**Discussion by amir-rakhimov** [nig-sc/nigsc_homepage2#395] [Q&A] [answered]
Hi everyone,

I would like to use MetaCache for analysing metagenomics data on the NIG supercomputer. The tool uses GPUs and requires installation of `zlib1g` through `sudo` as described here: [metacache/docs/gpu_version.md at master · muellan/metacache · GitHub](https://github.com/muellan/metacache/blob/master/docs/gpu_version.md). Since I don't have admin privileges, I would like to know if the administrators can install this library on NIG supercomputer.

There are a few other requirements on the webpage I sent. Can I install all the software myself? Also, it says that for the default database, MetaCache needs 120 GB of GPU memory (e.g. 4x GPUs with 32 GB each). If I remember correctly, the NIG supercomputer has 64 GPUs (https://sc.ddbj.nig.ac.jp/en/guides/hardware). But what is the total memory?

**Comment by oogasawa** [comment, marked as answer]
1. Maintenance of the supercomputer will be performed in time for the
statutory power outage at the NIG on 11/25. We will replace the OS
with Ubuntu Linux 22.04LTS at that time, and expand the libraries and
tools along with the OS replacement. At that time, zlib1g will also be
installed. (If not, please point it out to me.)

Please refer to the following announcement for details.
https://sc.ddbj.nig.ac.jp/en/blog/2023-11-24-scheduled-maintenance/

2.
> MetaCache needs 120 GB of GPU memory (e.g. 4x GPUs with 32 GB each)

The GPU node of the NIG supercomputer has four NVIDIA Tesla V100 SXM2
in one chassis, eachNVIDIA Tesla V100 SXM2 has 16 GB of memory.
Therefore, each GPU node has 64 GB of GPU memory.
If 120 GB of GPU memory is required, the calculation would not be
feasible on this type of GPU node.

However, when I check the metacache homepage,
I see that if you don't have enough RAM, you can use database partitioning.
https://github.com/muellan/metacache/blob/master/docs/gpu_version.md

(The link is broken, but I guess it points here.
https://github.com/muellan/metacache/blob/master/docs/partitioning.md )

We apologize for the inconvenience, but could you please run this
through the database partitioning?

**Reply by amir-rakhimov** [reply]
Thanks for the response. I want to clarify two details: 

1. Only the Type 2b node has GPUs, right? 
2. And since we have four NVIDIA Tesla V100 SXM2 with 16 GB of memory each, the total GPU memory on the supercomputer is 64GB?

**Reply by oogasawa** [reply]
> 1. Only the Type 2b node has GPUs, right?

Yes, that's correct. (GPU node = Type 2b node)

> 2. And since we have four NVIDIA Tesla V100 SXM2 with 16 GB of memory each, the total GPU memory on the supercomputer is 64GB?

Yes, that's correct.
The total GPU memory on each Type 2b node is 16GB x 4 = 64GB.

There are 16 GPU nodes (= Type 2b nodes) in the NIG supercomputer. (64GB x 16 = 1,024GB GPU memory)

**Reply by amir-rakhimov** [reply]
Ok, understood. But if we have 1024 GB of memory, that means it's possible to run MetaCache without partitioning? It'll take two Type 2b nodes. Please correct me if I'm wrong cause I'm not familiar with the Supercomputer architecture 😅

**Reply by oogasawa** [reply]
In order to achieve parallelization across computation nodes, the program must be written in slightly specialized code using MPI or similar technology. Also, the interconnect network between nodes is slower than the communication between GPUs within a node. I believe the requirement on the total amount of GPU memory mentioned in the metacache documentation is for a single compute node, and metacache does not support parallelization using multiple computers at the same time.
